libnvme/libnvme/libnvme_ctrl_spec.h

```diff
@@ -0,0 +1,1498 @@
+ *	uint16_t.
+ */
+_DLL_PUBLIC uint16_t nvme_ctrl_wctemp_get(struct nvme_ctrl *cnt);
+
+/**
+ * nvme_ctrl_cctemp_get() - Retrieve the CCTEMP property of specified
+ * NVMe controller.
+ *
+ * Retrieve the CCTEMP(Critical Composite Temperature Threshold) property of
+ * specified NVMe controller.
+ *
+ * @cnt:
+ *	Pointer of 'struct nvme_ctrl'.
+ *	If this pointer is NULL, your program will be terminated by assert().
+ *
+ * Return:
+ *	uint16_t.
+ */
+_DLL_PUBLIC uint16_t nvme_ctrl_cctemp_get(struct nvme_ctrl *cnt);
+
+/**
+ * nvme_ctrl_mtfa_get() - Retrieve the MTFA property of specified
+ * NVMe controller.
+ *
+ * Retrieve the MTFA(Maximum Time for Firmware Activation) property of specified
+ * NVMe controller.
+ *
+ * @cnt:
+ *	Pointer of 'struct nvme_ctrl'.
+ *	If this pointer is NULL, your program will be terminated by assert().
+ *
+ * Return:
+ *	uint16_t.
+ */
+_DLL_PUBLIC uint16_t nvme_ctrl_mtfa_get(struct nvme_ctrl *cnt);
+
+/**
+ * nvme_ctrl_hmpre_get() - Retrieve the HMPRE property of specified
+ * NVMe controller.
+ *
+ * Retrieve the HMPRE(Host Memory Buffer Preferred Size) property of specified
+ * NVMe controller.
+ *
+ * @cnt:
+ *	Pointer of 'struct nvme_ctrl'.
+ *	If this pointer is NULL, your program will be terminated by assert().
+ *
+ * Return:
+ *	uint32_t.
+ */
+_DLL_PUBLIC uint32_t nvme_ctrl_hmpre_get(struct nvme_ctrl *cnt);
+
+/**
+ * nvme_ctrl_hmmin_get() - Retrieve the HMMIN property of specified
+ * NVMe controller.
+ *
+ * Retrieve the HMMIN(Host Memory Buffer Minimum Size) property of specified
+ * NVMe controller.
+ *
+ * @cnt:
+ *	Pointer of 'struct nvme_ctrl'.
+ *	If this pointer is NULL, your program will be terminated by assert().
+ *
+ * Return:
+ *	uint32_t.
+ */
+_DLL_PUBLIC uint32_t nvme_ctrl_hmmin_get(struct nvme_ctrl *cnt);
+
+/**
+ * nvme_ctrl_tnvmcap_get() - Retrieve the TNVMCAP property of specified
+ * NVMe controller.
+ *
+ * Retrieve the TNVMCAP(Total NVM Capacity) property of specified NVMe
+ * controller.
+ * You may use these lines to retrieve the least significant 64 bits of it:
+ *
+ *	le64toh(*((uint64_t *) nvme_ctrl_tnvmcap_get(cnt);
+ *
+ * @cnt:
+ *	Pointer of 'struct nvme_ctrl'.
+ *	If this pointer is NULL, your program will be terminated by assert().
+ *
+ * Return:
+ *	uint8_t array with size of 16 representing a 128 bit integer in little
+ *	endian.
+ */
+_DLL_PUBLIC const uint8_t *nvme_ctrl_tnvmcap_get(struct nvme_ctrl *cnt);
+
+/**
+ * nvme_ctrl_unvmcap_get() - Retrieve the UNVMCAP property of specified
+ * NVMe controller.
+ *
+ * Retrieve the UNVMCAP(Unallocated NVM Capacity) property of specified NVMe
+ * controller.
+ * You may use these lines to retrieve the least significant 64 bits of it:
+ *
+ *	le64toh(*((uint64_t *) nvme_ctrl_unvmcap_get(cnt);
+ *
+ * @cnt:
+ *	Pointer of 'struct nvme_ctrl'.
+ *	If this pointer is NULL, your program will be terminated by assert().
+ *
+ * Return:
+ *	uint8_t array with size of 16 representing a 128 bit integer in little
+ *	endian.
```

Comment:
Need to clarify/document with the caller about the lifetime of the memory returned, eg. should it be free'd etc.